backend/tests/v5/unit/schema/tickets/templates.test.js

```diff
@@ -0,0 +1,296 @@
+/**
+ *  Copyright (C) 2022 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+const { cloneDeep } = require('lodash');
+const { src } = require('../../../helper/path');
+const { generateRandomString } = require('../../../helper/services');
+
+const TemplateSchema = require(`${src}/schemas/tickets/templates`);
+const { fieldTypes, presetModules, presetEnumValues, presetModulesProperties, defaultProperties } = require(`${src}/schemas/tickets/templates.constants`);
+
+const testValidate = () => {
+	const nameTests = [
+		['the name is too long', { name: generateRandomString(121), code: generateRandomString(3) }, false],
+		['the name is an empty string', { name: '', code: generateRandomString(3) }, false],
+		['the name is not defined', { code: generateRandomString(3) }, false],
+	];
+
+	const codeTests = [
+		['code is not defined', { name: generateRandomString() }, false],
+		['code is too long', { name: '', code: generateRandomString(4) }, false],
+		['code is too short', { name: '', code: generateRandomString(2) }, false],
+	];
+
+	const schemaFieldsTest = [
+		['all optional fields provided', {
+			name: generateRandomString(),
+			code: generateRandomString(3),
+			config: {
+				comments: false,
+				issueProperties: true,
+				defaultView: true,
+				defaultImage: false,
+			},
+			deprecated: true,
+			properties: undefined,
+			modules: undefined,
+		}, true],
+		['properties is an empty array', { name: generateRandomString(), code: generateRandomString(3), properties: [] }, true],
+		['properties is of the wrong type', { name: generateRandomString(), code: generateRandomString(3), properties: 'a' }, false],
+		['property name is used by a default property', { name: generateRandomString(), code: generateRandomString(3), properties: [defaultProperties[0]] }, false],
+		['modules is an empty array', { name: generateRandomString(), code: generateRandomString(3), modules: [] }, true],
+		['modules is of the wrong type', { name: generateRandomString(), code: generateRandomString(3), modules: 'a' }, false],
+	];
+
+	const propertiesTest = [
```

Comment:
we could add a check for two identical values

Reply:
is this not it? 
![image](https://user-images.githubusercontent.com/11945337/184387604-27bbce9a-f323-4b84-a49d-81b8c91121cf.png)